src/setting_sbp_cb.c

```diff
@@ -116,7 +128,14 @@ static void setting_register_resp_callback(uint16_t sender_id,
     return;
   }
 
-  setting_update_value(ctx, (uint8_t *)resp->setting, len - sizeof(resp->status));
+  /* Update the actual setting. In case of readonly, trust the initialized value. */
+  setting_value_update(ctx,
+                       resp->setting,
+                       len - sizeof(resp->status),
+                       UPDATE_FILTER_WATCHONLY | UPDATE_FILTER_READONLY);
+
+  /* Update watchers, no need for filter as wathers shall not be read-only */
+  setting_value_update(ctx, resp->setting, len - sizeof(resp->status), UPDATE_FILTER_NONE);
```

Comment:
not understanding maybe, but why the double call? the only difference I see between the two is early return if either read only or watcher, but then the second call would follow the logic previously avoided with early returns?

Reply:
True, this is a blunder based on earlier version.